### PR TITLE
fix(se): Empty dbgenerated() breaking for Unsupported() types

### PR DESCRIPTION
**Pull request by Druue**
fixes [#15654](https://github.com/prisma/prisma/issues/15654)

### CODE REVIEW

File: schema-engine/sql-schema-describer/src/lib.rs [hunk 1]
```diff
@@ -825,8 +825,8 @@ pub enum DefaultKind {
 }
 
 impl DefaultValue {
-    pub fn db_generated(val: impl Into<String>) -> Self {
-        Self::new(DefaultKind::DbGenerated(Some(val.into())))
+    pub fn db_generated<S: Into<String>>(val: impl Into<Option<S>>) -> Self {
```
Comment (jkomyno) [inline schema-engine/sql-schema-describer/src/lib.rs:828]: There's already an `Into` in the generic parameter `<S: Into<String>>`, so I wonder why we need `val: impl Into<Option<S>>` as well? Wouldn't it be sufficient to have `val: Option<S>` or something like it?

**Review by jkomyno — APPROVED** [review]
Approved with a small nit

File: schema-engine/sql-schema-describer/src/lib.rs (hunk 1 above)
Comment (Druue) [reply schema-engine/sql-schema-describer/src/lib.rs:828]: The issue here is the difference in what we provide from the `unsupported` path vs for regular `PrismaValue`s. Having both `Into`s allowed us to change the function to accept what we provide from both paths without needing to change the args we pass everywhere it's used in our tests

Comment (jkomyno) [reply schema-engine/sql-schema-describer/src/lib.rs:828]: Thanks for the additional context! Feel free to merge

**Pull request merged**